src/main/resources/db/migration/V1__analysis.sql

```diff
@@ -18,8 +18,8 @@ CREATE TABLE repo_metrics
 
     CONSTRAINT unique_composite_key
         UNIQUE (repo_id, metric_language_id),
-    FOREIGN KEY (repo_id) REFERENCES repo(id),
-    FOREIGN KEY (metric_language_id) REFERENCES metric_language(id),
+    FOREIGN KEY (repo_id) REFERENCES repo(id) ON DELETE CASCADE,
+    FOREIGN KEY (metric_language_id) REFERENCES metric_language(id) ON DELETE CASCADE,
```

Comment:
Be careful when cascading deletes across many-to-many relationships. If two rows of the left table both point to the same row in the right table, then cascading will delete the right entry if either one of the left entries is deleted. I would refrain from using cascading altogether. Better to be left with a reusable orphan than deleting more than is needed.

Reply:
Also, DO NOT EDIT ALREADY EXECUTED MIGRATION FILES!

Reply:
loud and clear